internal/querycoordv2/task/scheduler.go

```diff
@@ -527,7 +547,28 @@ func (scheduler *taskScheduler) process(task Task) bool {
 		zap.Int64("source", task.SourceID()),
 	)
 
-	if !scheduler.executor.Exist(task.ID()) && task.IsFinished(scheduler.distMgr) {
+	actions, step := task.Actions(), task.Step()
+	for step < len(actions) && actions[step].IsFinished(scheduler.distMgr) {
+		task.StepUp()
+		step++
+	}
+
+	if step == len(actions) {
+		step--
+	}
+
+	executor, ok := scheduler.executors[actions[step].Node()]
```

Comment:
what will happen if the task executor has been removed?
wait for task timeout?

Reply:
all tasks related to the node will be stale, and then removed, `checkStale()` checks whether the node is offline per scheduling round